forks/Microsoft.Health.FellowOakDicom/Core/DicomElement.cs

```diff
@@ -262,14 +262,27 @@ public override T Get<T>(int item = -1)
             if (item == -1) item = 0;
             if (item < 0 || item >= Count) throw new ArgumentOutOfRangeException(nameof(item), "Index is outside the range of available value items");
 
-            // If nullable, need to apply conversions on underlying type (#212)
-            var t = Nullable.GetUnderlyingType(typeof(T)) ?? typeof(T);
-
-            return (T)Convert.ChangeType(_values[item], t);
+            return GetValue<T>(_values[item]);
         }
 
         return base.Get<T>(item);
     }
 
     #endregion
+
+    #region Private Members
+
+    private T GetValue<T>(TType val)
+    {
+        // If nullable, need to apply conversions on underlying type (#212)
+        var t = Nullable.GetUnderlyingType(typeof(T)) ?? typeof(T);
+
+        if (!t.IsEnum) return (T)Convert.ChangeType(val, t);
+
+        if (Enum.IsDefined(typeof(T), val)) return (T)Enum.ToObject(t, val);
```

Comment:
`Enum.IsDefined` will throw an exception if the underlying `enum` type `T` and `val` are different types.

Reply:
Your best bet may be to simply always call `ToString()` for `enum` types and call `Parse`. I don't like that very much... but this is a test util in a fork lol

Reply:
Done.